crypto/x509/v3_attrdesc.c

```diff
@@ -0,0 +1,180 @@
+    if (i2a_ASN1_OBJECT(out, hash->algorithmIdentifier->algorithm) <= 0)
+        return 0;
+    if (BIO_puts(out, "\n") <= 0)
+        return 0;
+    if (hash->algorithmIdentifier->parameter) {
+        if (BIO_printf(out, "%*sParameter: ", indent, "") <= 0)
+            return 0;
+        if (ossl_print_attribute_value(out, 0, hash->algorithmIdentifier->parameter, indent + 4) <= 0)
+            return 0;
+        if (BIO_puts(out, "\n") <= 0)
+            return 0;
+    }
+    if (BIO_printf(out, "%*sHash Value: ", indent, "") <= 0)
+        return 0;
+    return print_hex(out, hash->hashValue->data, hash->hashValue->length);
+}
+
+static int i2r_INFO_SYNTAX_POINTER(X509V3_EXT_METHOD *method,
+                                   OSSL_INFO_SYNTAX_POINTER *pointer,
+                                   BIO *out, int indent) {
+    if (BIO_printf(out, "%*sNames:\n", indent, "") <= 0)
+        return 0;
+    if (OSSL_GENERAL_NAMES_print(out, pointer->name, indent) <= 0)
+        return 0;
+    if (BIO_puts(out, "\n") <= 0)
+        return 0;
+    if (pointer->hash != NULL) {
+        if (BIO_printf(out, "%*sHash:\n", indent, "") <= 0)
+            return 0;
+        if (i2r_HASH(method, pointer->hash, out, indent + 4) <= 0)
+            return 0;
+    }
+    return 1;
+}
+
+static int i2r_OSSL_INFO_SYNTAX(X509V3_EXT_METHOD *method,
+                                OSSL_INFO_SYNTAX *info,
+                                BIO *out, int indent) {
+    switch (info->type) {
+    case (OSSL_INFO_SYNTAX_TYPE_CONTENT): {
+        if (BIO_printf(out, "%*sContent: ", indent, "") <= 0)
+            return 0;
+        if (BIO_printf(out, "%.*s", info->choice.content->length, info->choice.content->data) <= 0)
+            return 0;
+        if (BIO_puts(out, "\n") <= 0)
+            return 0;
+        return 1;
+    }
+    case (OSSL_INFO_SYNTAX_TYPE_POINTER): {
+        if (BIO_printf(out, "%*sPointer:\n", indent, "") <= 0) {
+            return 0;
+        }
+        return i2r_INFO_SYNTAX_POINTER(method, info->choice.pointer, out, indent + 4);
+    }
+    default: return 0;
+    }
+    return 0;
+}
+
+static int i2r_OSSL_PRIVILEGE_POLICY_ID(X509V3_EXT_METHOD *method,
+                                   OSSL_PRIVILEGE_POLICY_ID *ppid,
+                                   BIO *out, int indent) {
+    char buf[80];
+
+    /* Intentionally display the numeric OID, rather than the textual name. */
+    if (OBJ_obj2txt(buf, sizeof(buf), ppid->privilegePolicy, 1) <= 0)
+        return 0;
+    if (BIO_printf(out, "%*sIdentifier: %s\n", indent, "", buf) <= 0)
+        return 0;
+    if (BIO_printf(out, "%*sSyntax:\n", indent, "") <= 0)
+        return 0;
+    return i2r_OSSL_INFO_SYNTAX(method, ppid->privPolSyntax, out, indent + 4);
+}
+
+static int i2r_OSSL_ATTRIBUTE_DESCRIPTOR(X509V3_EXT_METHOD *method,
+                                         OSSL_ATTRIBUTE_DESCRIPTOR *ad,
+                                         BIO *out, int indent) {
+    char buf[80];
```

Comment:
are we guaranteed to have an object identifier less than 80 chars?  Might it be better to use a BIO_s_mem here?

Reply:
I just copied the 80-bytes standard from here: https://github.com/openssl/openssl/blob/50e9d2b188b8dce070f388640c06a7dc04417390/crypto/asn1/tasn_prn.c#L404-L415

But I would be happy to do BIO_s_mem instead.

Reply:
@nhorman or perhaps @t8m do you have an opinion on the above?

Reply:
@nhorman if you grep "[80]" in the codebase, you will find that it is a widespread practice for query SN/LN names from OIDs.

Reply:
@nhorman can I mark this as resolved?

Reply:
If 80 chars is the de-facto standard in our code, then I'm fine with it, though I'd be curious to know if thats codified in a standard anywhere.  Thats a future question though, we can resolve this